datalad/support/gitrepo.py

```diff
@@ -214,8 +253,12 @@ def _normalize_path(base_dir, path):
     return relpath(path, start=base_dir)
 
 
+class _WithPath(Protocol):
+    path: str
```

Comment:
why not to move it to `datalad/typing.py` as plain `WithPath` and may be even `WithPathT` since it seems you use those whenever construct is not clearly a type, but could be some class.

Reply:
Are there functions in other files that operate on any type that has a `path` attribute?

Reply:
I would guess so, in particular in `annexrepo.py` since that one is close companion of this one. But also lots of files listed in hit for `git grep -l '\.path\>' `

Reply:
* There are only two top-level functions in `annexrepo.py`, and they both operate on IO handles rather than `WithPath`.
* A large number of hits for that regex doesn't mean that `WithPath` will be useful; most of those can be expected to operate on specific classes.